tests/test_tokenization_common.py

```diff
@@ -1153,6 +1153,120 @@ def test_chat_template_batched(self):
                     dummy_conversations, chat_template=dummy_template, tokenize=True
                 )  # Check that no error raised
 
+    @require_jinja
+    def test_chat_template_return_labels(self):
+        dummy_template = (
+            "{% for message in messages %}"
+            "{% if (message['role'] != 'assistant') %}"
+            "{{'<|im_start|>' + message['role'] + '\n' + message['content'] + '<|im_end|>' + '\n'}}"
+            "{% elif (message['role'] == 'assistant')%}"
+            "{{'<|im_start|>' + message['role'] + '\n'}}"
+            "{% generation %}"
+            "{{message['content'] + '<|im_end|>'}}"
+            "{% endgeneration %}"
+            "{{'\n'}}"
+            "{% endif %}"
+            "{% endfor %}"
+        )
+        conversations = [
+            [
+                {"role": "system", "content": "system message"},
+                {"role": "user", "content": "user message"},
+                {"role": "assistant", "content": "<start> assistant message 5"},
+                {"role": "user", "content": "user message 2"},
+                {"role": "assistant", "content": "<start2> assistant message 6"},
+            ],
+            [
+                {"role": "system", "content": "system message 3"},
+                {"role": "user", "content": "user message 3"},
+                {"role": "assistant", "content": "<start3> assistant message 3 7"},
+                {"role": "user", "content": "user message 4"},
+                {"role": "assistant", "content": "<start4> assistant message 4 8"},
+            ],
+        ]
+        assistant_wrappers = [
+            [("<start>", "5<|im_end|>"), ("<start2>", "6<|im_end|>")],
+            [("<start3>", "7<|im_end|>"), ("<start4>", "8<|im_end|>")],
+        ]
+        for tokenizer, pretrained_name, _ in self.tokenizers_list:
```

Comment:
This logic is redundant - for models which have a rust tokenizer, `self.tokenizers_list` will have two elements, so the logic below is run twice